catalog/catalog.go

```diff
@@ -0,0 +1,64 @@
+package catalog
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/ghodss/yaml"
+	"github.com/giantswarm/microerror"
+	"github.com/levigross/grequests"
+)
+
+// GetLatestChart returns the latest chart tarball file in the specified catalog.
+func GetLatestChart(ctx context.Context, catalog, app string) (string, error) {
+	index, err := getIndex(catalog)
+	if err != nil {
+		return "", microerror.Mask(err)
+	}
+
+	var downloadURL string
+	{
+		entry, ok := index.Entries[app]
+		if !ok {
+			return "", microerror.Maskf(notFoundError, fmt.Sprintf("no app %q in index.yaml", app))
+		}
+		downloadURL = entry[0].Urls[0]
```

Comment:
Not sure why `URL` have to be array type, but using the first one guarantee that is the latest one.

Reply:
I think this is related to helm chart, https://github.com/giantswarm/default-catalog/blob/fc28e7623f150aa3598b9275acd7619a71209fa6/index.yaml#L18

Reply:
Yes good point. I was forgetting this is chart repository stuff and we just host on GitHub.

https://helm.sh/docs/chart_repository/